sdk/src/client/stronghold/migration.rs

```diff
@@ -12,17 +12,20 @@ use iota_stronghold::{Client, SnapshotPath, Stronghold};
 use zeroize::Zeroize;
 
 use super::{common::PRIVATE_DATA_CLIENT_PATH, Error, StrongholdAdapter};
-use crate::client::stronghold::{check_or_create_snapshot, storage::insert as v3_insert, Error as StrongholdError};
+use crate::client::{
+    secret::types::Password,
+    stronghold::{check_or_create_snapshot, storage::insert as v3_insert, Error as StrongholdError},
+};
 
 impl StrongholdAdapter {
     /// Migrates a snapshot from version 2 to version 3.
-    pub fn migrate_snapshot_v2_to_v3<P: AsRef<Path>>(
+    pub fn migrate_snapshot_v2_to_v3<P: AsRef<Path>, PW: Into<Password>>(
```

Comment:
Using a single generic forces the two passwords to be the same type, FYI

Reply:
yeah, that's good to know. I actually ran into the problem at first, then realized why it was done this way with the path.